numpy/_core/tests/test_multithreading.py

```diff
@@ -271,3 +271,19 @@ def closure(b):
         # Reducing the number of threads means the test doesn't trigger the
         # bug. Better to skip on some platforms than add a useless test.
         pytest.skip("Couldn't spawn enough threads to run the test")
+
+@pytest.mark.parametrize("dtype", [bool, int, float])
+def test_nonzero_bool(dtype):
+    # np.nonzero uses np.count_nonzero to determine the size of the output array
+    # In a second pass the indices of the non-zero elements are determined, but they can have changed
+    x = np.random.randint(4, size=10_000).astype(dtype)
+
+    def func(seed):
+        x[::2] = np.random.randint(2)
```

Comment:
I'm not actually sure what the best thing to do here is. It's probably better for each thread to get its own private RNG so there's no cross-talk or shared RNG state. I know the new numpy RNG infrastructure had a lot of careful thought put into it to make it possible to do stuff like this. 

Maybe @rkern has a suggestion here?

Reply:
I do not think the randomization is really important, we could also just flip between many non-zeros and many zeros in the array. I removed the seed argument